commit 7293795907ee87bda82342e6b37812ab671455ab
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Tue Jul 3 14:23:47 2018 +0100

    spi: Make GPIO CSs honour the SPI_NO_CS flag
    
    The SPI configuration state includes an SPI_NO_CS flag that disables
    all CS line manipulation, for applications that want to manage their
    own chip selects. However, this flag is ignored by the GPIO CS code
    in the SPI framework.
    
    Correct this omission with a trivial patch.
    
    See: https://github.com/raspberrypi/linux/issues/2169
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/spi/spi.c b/drivers/spi/spi.c
index 9da0bc5a036c..f611024c5e5e 100644
--- a/drivers/spi/spi.c
+++ b/drivers/spi/spi.c
@@ -733,7 +733,9 @@ static void spi_set_cs(struct spi_device *spi, bool enable)
 		enable = !enable;
 
 	if (gpio_is_valid(spi->cs_gpio)) {
-		gpio_set_value(spi->cs_gpio, !enable);
+		/* Honour the SPI_NO_CS flag */
+		if (!(spi->mode & SPI_NO_CS))
+			gpio_set_value(spi->cs_gpio, !enable);
 		/* Some SPI masters need both GPIO CS & slave_select */
 		if ((spi->controller->flags & SPI_MASTER_GPIO_SS) &&
 		    spi->controller->set_cs)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1305_7293795907ee87bda82342e6b37812ab671455ab.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1305_7293795907ee87bda82342e6b37812ab671455ab.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
